modules/dr-recover-lost-control-plane-hosts.adoc

```diff
@@ -0,0 +1,301 @@
+2019-05-15 19:03:34.647589 I | pkg/netutil: resolving etcd-0.clustername.devcluster.openshift.com:2380 to 10.0.143.125:2380
+2019-05-15 19:03:34.883545 I | mvcc: restore compact to 361491
+2019-05-15 19:03:34.915679 I | etcdserver/membership: added member cbe982c74cbb42f [https://etcd-0.clustername.devcluster.openshift.com:2380] to cluster 807ae3bffc8d69ca
+Starting static pods..
+..starting kube-scheduler-pod.yaml
+..starting kube-controller-manager-pod.yaml
+..starting kube-apiserver-pod.yaml
+..starting etcd-member.yaml
+Starting kubelet..
+----
++
+Once the `etcd-snapshot-restore.sh` script completes, your cluster should now have a single member
+etcd cluster, and API services will begin restarting. This might take up to 15 minutes.
++
+In a terminal that has access to the cluster, run the following command to verify that it is ready:
++
+----
+$ oc get nodes -l node-role.kubernetes.io/master
+NAME                                         STATUS   ROLES    AGE   VERSION
+ip-10-0-143-125.us-east-2.compute.internal   Ready    master   46m   v1.13.4+db7b699c3
+----
++
+[NOTE]
+====
+Be sure that all old etcd members being replaced are shut down. Otherwise, they might try to connect to the new cluster and will report errors like the following in the logs:
+----
+2019-05-20 15:33:17.648445 E | rafthttp: request cluster ID mismatch (got 9f5f9f05e4d43b7f want 807ae3bffc8d69ca)
+----
+====
+
+. Create new master hosts.
++
+If your cluster has its Machine API enabled and functional, then when the
+OpenShift `machine-api` Operator is restored, it will create the new masters. If you do
+not have the `machine-api` Operator enabled, you must create new masters
+using the same methods that were used to originally create them.
++
+In a terminal that has access to the cluster, run the following command to verify that your masters are ready:
++
+----
+$ oc get nodes -l node-role.kubernetes.io/master
+NAME                                         STATUS   ROLES    AGE   VERSION
+ip-10-0-143-125.us-east-2.compute.internal   Ready    master   50m   v1.13.4+db7b699c3
+ip-10-0-156-255.us-east-2.compute.internal   Ready    master   92s   v1.13.4+db7b699c3
+ip-10-0-162-178.us-east-2.compute.internal   Ready    master   70s   v1.13.4+db7b699c3
+----
+
+. Correct the DNS entries.
+
+.. Obtain the private IP address of the new instance.
++
+In a terminal that has access to the cluster, run the following command:
++
+----
+$ oc get node ip-10-0-143-125.us-east-2.compute.internal -o  jsonpath='{.status.addresses[?(@.type=="InternalIP")].address}{"\n"}'
+10.0.143.125
+----
+
+.. From the AWS console, edit the etcd-0 Route 53 record in the private DNS zone and set the value to the new private IP address. See link:https://docs.aws.amazon.com/Route53/latest/DeveloperGuide/resource-record-sets-editing.html[Editing Records] in the AWS documentation for instructions.
+
+.. Repeat these steps to update the etcd-1 and etcd-2 entries as appropriate to the current addresses.
```

Comment:
@hexfusion Trying to wrap up our convo earlier on this. So if we assume in this procedure that you had just the one master that you restored and are replacing the others, is this step still appropriate here or not? Per item 2 in https://bugzilla.redhat.com/show_bug.cgi?id=1711891

If so, do we need to call out here that this step is only appropriate in that case (and not if you're just replacing a single master, but the majority of your masters were fine)?

Reply:
In this scenario, we are restoring from majority failure which would assume in a 3 node cluster 2 nodes are NLA. In which case we would need to update DNS to properly reflect new master nodes once they are up. I would assume because the remaining node is still alive that its DNS is OK. After the new machines are created we need to make sure to update the A records to properly reflect.

When all new masters are online and before they are added to the cluster A records should be verified/updated for each new. master node.

Reply:
It does not hurt to verify all DNS A records.